release-versions/generate_release_file.py

```diff
@@ -25,49 +27,36 @@ def get_versions(repository_name):
-
-        intg_date = get_date_for_stage(repository_name, intg_sha)
-        staging_date = get_date_for_stage(repository_name, staging_sha)
-        prod_date = get_date_for_stage(repository_name, prod_sha)
-
-        max_version = max([intg_version, staging_version, prod_version])
-
-        staging_out_of_date = staging_version != max_version
-        prod_out_of_date = prod_version != max_version
-
-        staging_diff = (intg_date - staging_date).days
-        prod_diff = (staging_date - prod_date).days
-
-        staging_diff_text = f"({staging_diff} day{'' if staging_diff == 1 else 's'} behind integration)" if staging_out_of_date else ""
-        prod_diff_text = f"({prod_diff} day{'' if prod_diff == 1 else 's'} behind staging)" if prod_out_of_date else ""
-
-        return {"repository": repository_name,
-                "integration": {"version": intg_version,
-                                "data_class":
-                                    "table-success" if intg_version == max_version else "table-danger fw-bold",
-                                "out_of_date": intg_version != max_version,
-                                "date_diff": ""
-                                },
-                "staging": {"version": staging_version,
-                            "data_class":
-                                "table-success" if staging_version == max_version else "table-danger fw-bold",
-                            "out_of_date": staging_out_of_date,
-                            "date_diff": staging_diff_text
-                            },
-                "production": {"version": prod_version,
-                               "data_class":
-                                   "table-success" if prod_version == max_version else "table-danger fw-bold",
-                               "out_of_date": prod_out_of_date,
-                               "date_diff": prod_diff_text
-                               },
-                }
+        intg_branch = release_branch(repository_name, "intg", filtered_release_branches, tags)
+        staging_branch = release_branch(repository_name, "staging", filtered_release_branches, tags)
+        prod_branch = release_branch(repository_name, "prod", filtered_release_branches, tags)
+
+        defined_versions = [branch["version"] for branch in [intg_branch, staging_branch, prod_branch] if branch]
+        max_version = max(defined_versions)
+
+        return {
+            "repository": repository_name,
+            "integration": integration_view_model(intg_branch, max_version),
+            "staging": higher_environment_branch_view_model(staging_branch, intg_branch, max_version),
+            "production": higher_environment_branch_view_model(prod_branch, staging_branch, max_version),
+        }
+
+
+def release_branch(repo_name, environment, filtered_release_branches, repo_tags):
+    branch = filtered_release_branches.get(f"release-{environment}")
+
+    if (branch):
```

Comment:
Very minor but the Python linter complains about brackets around the if statement. There's another one on line 88 as well.

Reply:
Fixed, thanks